### PR TITLE
DR-3171 log Terrra user id to BigQuery but not MixPanel

### PR DESCRIPTION
**Pull request by nmalfroy**
Though explicitly: do not log it to MixPanel.
This will allow tying to Terra users in the data warehouse (which today isn't possible since the `distinct_id` is a MixPanel user id.

This ends up logging events that look something like:
```
{
  event: 'foo',
  properties: {
    appId: 'myapp',
    someprop: 'this prop',
    pushToMixpanel: false,
    token: null,
    distinct_id: '<some id.  Maybe a terra user id, maybe not>',
    terra_user_id: '<the terra user id>'
  }
}
```

Also very much of note in this PR: this PR introduces Jest and unit tests for all endpoints in Bard.  A run looks like:
```
wm6b3-2bb:bard nmalfroy$ yarn test
yarn run v1.22.0
$ jest --detectOpenHandles --forceExit
 PASS  src/app.test.js
  Test the root path
    ✓ calling status returns 200 (104 ms)
  Test static docs
    ✓ calling / redirects and returns the swagger page (30 ms)
    ✓ calling docs shows apidocs page (39 ms)
    ✓ calling bogus url fails with not found error (27 ms)
  Test sending events
    ✓ calling event with post happy path (unauthed) (69 ms)
    ✓ calling event with post happy path (unauthed) - no mixpanel (29 ms)
    ✓ calling event with post happy path (authed) (29 ms)
    ✓ calling event with get fails (28 ms)
    ✓ calling event with post fails with no properties (34 ms)
    ✓ calling event with post fails with no event (27 ms)
    ✓ calling event with post fails with insufficient properties (30 ms)
    ✓ calling event with post fails with invalid distinct_i (30 ms)
    ✓ calling event with post fails with distinct_id and user (28 ms)
    ✓ calling event with post fails with unauthed user (29 ms)
  Test identifying users
    ✓ calling identify user happy path (30 ms)
    ✓ calling identify user fails when user disabled (26 ms)
    ✓ calling identify user fails with no anonId (29 ms)
  Test syncing profiles
    ✓ calling syncProfile happy path (34 ms)
    ✓ calling identify user fails when user disabled (29 ms)

Test Suites: 1 passed, 1 total
Tests:       19 passed, 19 total
Snapshots:   0 total
Time:        2.332 s, estimated 3 s
Ran all test suites.
✨  Done in 3.74s.
```

Tests can be run interactively as well.

In order to test, I ended up needing to pull server.js into a bunch of different subcomponent files.

One testing note: I mostly couldn't get spying to work (except for logging) so I just added explicit method call counters for the various fetch calls.

### CODE REVIEW

**Review by okotsopoulos — COMMENTED** [review]
Does Bard have a unit test suite (I think the answer is no)?  Can you share more about how how you tested this and verified behavior?

**Comment by nmalfroy** [comment]
> Does Bard have a unit test suite (I think the answer is no)? Can you share more about how how you tested this and verified behavior?

Great question.  There is currently no automated testing so I'll add that in this ticket.  I was able to test manually but I'll feel better with automated testing

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Note that there is currently no separate development environment for mixpanel, s
     yarn run lint
```
Comment (okotsopoulos) [inline README.md]: Would it be worth expanding the README to suggest running tests too?

**Review by okotsopoulos — APPROVED** [review]
Thank you *so* much for expanding the scope to include Bard unit tests.  I see that they run as part of the PR so a failure would show up there even if the developer doesn't run them locally.  This will make me feel better about reviewing Bard PRs in the future!

File: README.md (hunk 1 above)
Comment (nmalfroy) [reply README.md]: great call!  I'll add a section

Comment (nmalfroy) [reply README.md]: done!

**Review by samanehsan — APPROVED** [review]
Thanks for adding tests!

**Review by petesantos — APPROVED** [review]
Reviewed, ran tests, LGTM

**Pull request merged**